### PR TITLE
Add alt attributes and improve lighthouse scores

### LINKED ISSUE #33: Improve SEO score
https://pagespeed.web.dev/analysis/https-main--airindia--hlxsites-hlx-live-in-en-book-special-offers-global-connectivity/sttrcans3m?form_factor=mobile

image

### LINKED ISSUE #41: Improve accessibility and best practices - part 2
image

### LINKED ISSUE #42: Header anchor tags are missing href attributes(Desktop/Mobile)
Screenshot 2024-02-01 at 11 44 53 AM

### PR DESCRIPTION
**Pull request by adobenitesh**
Fix #33
Fix #41
Fix #42

Test URLs:
- Before: https://main--airindia--hlxsites.hlx.live/in/en/book/special-offers/global-connectivity
- After: https://i-33-41--airindia--hlxsites.hlx.live/in/en/book/special-offers/global-connectivity

### CODE REVIEW

File: blocks/header/header.js [hunk 1]
```diff
@@ -332,6 +332,22 @@ function addGlobalEventHandlers() {
...
+ * attach href attributes to Anchor tags
+ */
+
+function attachHrefToAnchorTags() {
+  const nav = document.getElementById('nav');
+  nav.querySelectorAll('a').forEach((anchor) => {
+    if (!anchor.getAttribute('href')) {
+      anchor.setAttribute('href', '#');
+    }
+    anchor.addEventListener('click', (e) => {
```
Comment (sselvara) [inline blocks/header/header.js:345]: @adobenitesh will this not prevent the redirection if the anchor link has url?

File: blocks/header/header.js [hunk 2]
```diff
@@ -354,9 +370,11 @@ export default async function decorate(block) {
   });
 
   const navBrand = nav.querySelector('.nav-brand');
-  const brandImg = navBrand.querySelector('img');
-  if (brandImg) {
-    brandImg.setAttribute('alt', 'Air India');
+  const brandImg = navBrand.querySelectorAll('img');
+  if (brandImg.length > 0) {
+    brandImg.forEach((img) => {
```
Comment (sselvara) [inline blocks/header/header.js:375]: Can't we add Alt value in the doc?

File: styles/styles.css [hunk 3]
```diff
@@ -145,6 +145,8 @@ body {
 
 * {
   box-sizing: border-box;
+  margin: 0;
```
Comment (sselvara) [inline styles/styles.css:148]: Please double check the entire page and make sure that adding global changes has no impact in other places.

**Review by sselvara — CHANGES_REQUESTED** [review]

**Comment by dipratap** [comment]
The footer docx needs changes for this PR, since main is dependent on footer docx changes, I haven't published the footer changes yet (they are saved though in the docx so if previewed they will appear) so the footer in https://i-33-41--airindia--hlxsites.hlx.live/in/en/book/special-offers/global-connectivity will appear distorted currently.

**Review by sdmcraft — COMMENTED** [review]
The footer looks broken. Not sure if its related to proposed changes but we should check 
image

**Comment by dipratap** [comment]
> The footer looks broken. Not sure if its related to proposed changes but we should check

@sdmcraft Quoting my earlier comment here:

> The footer docx needs changes for this PR, since main is dependent on footer docx changes, I haven't published the footer changes yet (they are saved though in the docx so if previewed they will appear) so the footer in https://i-33-41--airindia--hlxsites.hlx.live/in/en/book/special-offers/global-connectivity will appear distorted currently.

I can preview and publish the changes in footer doc after which the footer will look correct, but it will distort the main one until these PR changes are merged. Your thoughts?

**Comment by dipratap** [comment]
Temporary footer content docx added [here](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B753B48E5-2A98-4236-83F4-E38C13A11344%7D&file=footer.docx&action=default&mobileredirect=true) and used in this branch. The footer will appear correct now.

Original footer doc to be replaced with the temporary doc just before PR merge.

**Comment by dipratap** [comment]
@sselvara Could you please review this?

File: blocks/header/header.js [hunk 4]
```diff
@@ -375,6 +379,33 @@ function addGlobalEventHandlers() {
   }
   addAdobeLaunchLoadedHandler();
 }
+function addAltTextToHeaderIcons() {
+  // Add alt attribute to brand icon
+  const airIndiaIcon = document.querySelector('img[data-icon-name="ai-logo-white"]');
+  if (airIndiaIcon) {
+    airIndiaIcon.setAttribute('alt', 'Air India');
```
Comment (sselvara) [inline blocks/header/header.js:386]: can you please add these texts into placeholder file and fetch it to support i18n ?

File: blocks/header/header.js [hunk 5]
```diff
@@ -375,6 +379,33 @@ function addGlobalEventHandlers() {
...
+  }
+  // Add alt attribute to search icon
+  const searchIcon = document.querySelector('img[data-icon-name="search-light"]');
+  if (searchIcon) {
+    searchIcon.setAttribute('alt', 'Search icon');
```
Comment (sselvara) [inline blocks/header/header.js:391]: can you please add these texts into placeholder file and fetch it to support i18n ?

File: blocks/booking/booking.css [hunk 6]
```diff
@@ -2758,13 +2758,13 @@ body,html {
 }
 
 .booking-form .label,.booking-form label {
-    color: #8f959a;
+    color: #70777B;
```
Comment (sselvara) [inline blocks/booking/booking.css:2761]: use color variables

**Review by sselvara — CHANGES_REQUESTED** [review]

File: blocks/header/header.js (hunk 5 above)
Comment (dipratap) [reply blocks/header/header.js]: done

File: blocks/booking/booking.css (hunk 6 above)
Comment (dipratap) [reply blocks/booking/booking.css]: done

**Comment by dipratap** [comment]
TODO post PR approval:
- Change below to use '/footer' in footer.js

`const footerPath = footerMeta.footer || '/drafts/divyansh/footer';`

- Update the main footer.docx file with draft changes.

File: blocks/booking/booking.js [hunk 7]
```diff
@@ -14,6 +14,13 @@ const envDetails = {};
 
 let container;
 
+function updateDateButtonAriaLabels() {
+  const dateButtons = document.querySelectorAll('.btn.bi.bi-calendar3');
+  dateButtons.forEach((dateButton) => {
+    dateButton.setAttribute('aria-label', 'Date button');
```
Comment (sselvara) [inline blocks/booking/booking.js:20]: Use placeholder file to fetch the text

**Review by sselvara — CHANGES_REQUESTED** [review]

File: blocks/booking/booking.js (hunk 7 above)
Comment (dipratap) [reply blocks/booking/booking.js]: Sure, done

**Review by sselvara — APPROVED** [review]

**Pull request merged**